scripts/migrate.py

```
@@ -0,0 +1,193 @@
+            if not pending:
+                print("Nothing to do.")
+                return 0
+            for f in pending:
+                print(f"Applying {f.name}...", end=" ", flush=True)
+                sql = f.read_text()
+                checksum = file_checksum(f)
+                async with conn.transaction():
+                    await conn.execute(sql)
+                    await conn.execute(
+                        f"INSERT INTO {TRACKING_TABLE} (filename, checksum) "
+                        "VALUES ($1, $2)",
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)